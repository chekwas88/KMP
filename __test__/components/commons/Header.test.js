import React from "react";
import { Header } from "@components/commons/Header/index";
import { logo } from "@base/img/logo.png";

const props = {
  user: {
    isAuthenticated: true,
    username: "kingsmen"
  },
  profile: {
    firstname: "Kings",
    lastname: "men",
    avatar: logo
  }
};


const setup = () => {
  const wrapper = shallow(
    <Header {...props} />
  );
  return wrapper;
};

describe("<Header />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it("should show Author's Haven as title", () => {
    const header = wrapper.find("h1");
    expect(header.exists()).toBe(true);
    expect(header.text()).toEqual("Authorâ€˜s Haven");
  });

  it("should have login button in header", () => {
    const button = wrapper.find("Button").first();
    expect(button.exists()).toBe(true);
    expect(button.props().children).toEqual("Sign In");
  });

  it("should have logout button in header", () => {
    const button = wrapper.find("Button").at(1);
    expect(button.exists()).toBe(true);
    expect(button.props().children).toEqual("Get Started");
  });

  it("should render the Header component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find("Header")).toBeTruthy();
    wrapper.find("button").first;
  });
});
