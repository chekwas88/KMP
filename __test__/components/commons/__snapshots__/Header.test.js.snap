// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the Header component correctly 1`] = `
<Fragment>
  <div
    className="bg-white shadow relative"
  >
    <div
      className="container mx-auto px-4"
    >
      <div
        className="flex items-center justify-between py-4"
      >
        <div
          className="flex"
        >
          <Link
            to="/"
          >
            <img
              alt="Authors' Haven Icon"
              className="w-10 h-9 mr-2"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
          </Link>
          <h1
            className="font-bold text-xl sm-text-base"
          >
            Authorâ€˜s Haven
          </h1>
        </div>
        <div
          className="hidden sm:flex sm:items-center"
        >
          <Button
            color="blue"
            onClick={[Function]}
            stretch={false}
            type="regular"
          >
            Sign In
          </Button>
          <Button
            color="blue"
            onClick={[Function]}
            stretch={false}
            type="outlined"
          >
            Get Started
          </Button>
        </div>
        <div
          className="flex sm:hidden cursor-pointer"
        >
          <svg
            className="w-6 h-6 text-blue-600 cursor-pointer"
            onClick={[Function]}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M64 384h384v-42.666H64V384zm0-106.666h384v-42.667H64v42.667zM64 128v42.665h384V128H64z"
            />
          </svg>
        </div>
      </div>
      <div
        className="block sm:hidden bg-white border-t-2 py-2 hidden"
      >
        <div
          className="flex justify-between items-center pt-2"
        >
          <Button
            color="blue"
            onClick={[Function]}
            stretch={false}
            type="regular"
          >
            Sign In
          </Button>
          <Button
            color="blue"
            onClick={[Function]}
            stretch={false}
            type="outlined"
          >
            Get Started
          </Button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="block bg-white border-t-2 py-2 shadow-lg absolute right-0 w-full md:w-2/5 lg:w-2/5 hidden lg:max-w-xs"
  >
    <div
      className="tooltip container"
    />
    <div
      className="flex justify-between items-center pt-2 font-sans text-sm"
    />
  </div>
</Fragment>
`;
